lib/solveEquation/stepThrough.js

```diff
@@ -182,12 +297,35 @@ function step(equation, symbolName) {
 function addSimplificationSteps(steps, equation, debug=false) {
   let oldEquation = equation.clone();
 
+  /*
+    Collects substeps for factoring and simplifying the expression separately
+    because their logics are located in different places. Collect factoring
+    substeps first in case the original expression is factorable. If no factoring
+    substeps, proceed to collect simplifying substeps.
+    Some examples of when this comes into play:
+    e.g. x * x + 3x + 2 = 0 <- factor the left side
+    e.g. x + 4 + 2 = 0 <- simplify the left side
+  */
   const leftSteps = simplifyExpressionNode(equation.leftNode, false);
+  const simplifiedLeftNode = leftSteps.length !== 0
```

Comment:
Here you call `factor` on `simplifiedLeftNode` instead of `equation.leftNode`, but in the right hand side section, you call `factor` on `equation.RightNode`. Is there a reason for that difference? This is why I think there should be a helper function!!